src/main/java/com/example/petbuddybackend/service/notification/NotificationService.java

```diff
@@ -0,0 +1,60 @@
+package com.example.petbuddybackend.service.notification;
+
+import com.example.petbuddybackend.entity.notification.CaretakerNotification;
+import com.example.petbuddybackend.entity.notification.ClientNotification;
+import com.example.petbuddybackend.entity.user.Caretaker;
+import com.example.petbuddybackend.entity.user.Client;
+import com.example.petbuddybackend.repository.notification.CaretakerNotificationRepository;
+import com.example.petbuddybackend.repository.notification.ClientNotificationRepository;
+import com.example.petbuddybackend.repository.notification.NotificationRepository;
+import com.example.petbuddybackend.service.mapper.NotificationMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class NotificationService {
+
+    private final CaretakerNotificationRepository caretakerNotificationRepository;
+    private final ClientNotificationRepository clientNotificationRepository;
+    private final NotificationRepository notificationRepository;
+    private final WebsocketNotificationService websocketNotificationService;
+
+
+    public void addNotificationForCaretakerAndSend(Long objectId, String objectType, Caretaker caretaker, String message) {
+        CaretakerNotification notification = createCaretakerNotification(objectId, objectType, caretaker, message);
+        CaretakerNotification savedNotification = caretakerNotificationRepository.save(notification);
+        websocketNotificationService.sendNotification(
+                caretaker.getEmail(),
+                savedNotification
+        );
+    }
+
+    public void addNotificationForClientAndSend(Long objectId, String objectType, Client client, String message) {
+        ClientNotification notification = createClientNotification(objectId, objectType, client, message);
+        ClientNotification savedNotification = clientNotificationRepository.save(notification);
+        websocketNotificationService.sendNotification(
+                client.getEmail(),
+                savedNotification
+        );
+    }
+
```

Comment:
Wydaje mi się (a jeśli się mylę to mnie popraw), że tutaj powiadomienia polecą do użytkownika niezależnie czy jest zalogowany jako klient lub opiekun. Wydaje mi się to mylące

Reply:
Ew. można zostawić tak jak jest i frontend by dostał coś "nadprogramwego", tylko w powiadomieniu powinna być informacja dla jakiego profilu to idzie. Nie byłoby to błędne. Jakbyśmy robili taki port na telefony to nawet byłoby wskazane bo tam wszyskie powiadomienia byłyby wysyłane

Reply:
Tak więc myśle, ze mozna zostawic oba i jedynie dodac rzeczywiscie czy adresatem jest opiekun czy klient, wtedy nawet jak bedzie ten przycisk do przelaczenia profilu to np jesli ktos jest na opiekunie, bo nad przyciskiem "przelacz na klienta" mozna dac liczbe powiadomien